integration-libs/cdc/core/auth/services/user-authentication/cdc-user-authentication-token.service.ts

```diff
@@ -32,11 +32,11 @@ export class CdcUserAuthenticationTokenService {
       .set('client_id', this.authConfigService.getClientId())
       .set('client_secret', this.authConfigService.getClientSecret())
       .set('grant_type', 'custom')
-      .set('UID', UID)
-      .set('UIDSignature', UIDSignature)
-      .set('timeStamp', signatureTimestamp)
-      .set('idToken', idToken)
-      .set('baseSite', baseSite);
+      .set('UID', encodeURIComponent(UID))
+      .set('UIDSignature', encodeURIComponent(UIDSignature))
+      .set('signatureTimestamp', encodeURIComponent(signatureTimestamp))
+      .set('id_token', encodeURIComponent(idToken))
+      .set('baseSite', encodeURIComponent(baseSite));
```

Comment:
Is this change really related to this PR? Or it's unrelated bugfix? If so, should we backport it separtely also to previous versions of Spartacus (the latest 3.x and 4.x)?
I can see we do 2 things:
1. we uri-encode parameters values
2. we rename 2 parameres: `timeStamp` and `idToken`. What is the reason of renaming them?

Reply:
Hi Kris,

These changes were introduced as a part of the customer issue fix.
The backend API was changed to accept the parameter - `timeStamp` as `signatureTimestamp` and `idToken` as `id_token` while refactoring the backend code.

Reply:
This change seems to be a bugfix unrelated to the subject of this PR.
For clarity of commits and for the ease of possible backporting of the bugfix, what do you think about moving this change to a separate bugfix PR?

Btw. Do I understand correctly, that this change is a breaking change (as it changes the contract of properties sent to the backend)? If yes, then formally we should release it only in the major version (5.0). So then there will be no backporting to 3.x and 4.x, I guess.

Only customers having the latest backend version (containing the backend fix with the changed contract) will benefit from this frontend's change, but for others (using older backends), this frontend's change will completely break the CDC integration, right? If yes, I think we should document it somewhere in 5.0 migration docs - what is needed in backend to make CDC work.

Moreover, independently, for customers that will still use 3.x or 4.x for some time, I believe we will need some documentation how they can workaround the bug (by writing some code both on the frontend and backend manually). What do you think about it?

Reply:
Hi @Platonn,

Thanks for bringing this dicussion. This fix was introduced as a part of the customer ticket. The customer is on Spartacus 4.3. so these changes are backported to 4.x version currently.

We can think of backporting the fixes for 3.x if needed.

thanks.